x-pack/plugins/security_solution/public/common/components/threat_match/entry_item.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useCallback } from 'react';
+import { EuiFormRow, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import styled from 'styled-components';
+
+import { IFieldType, IIndexPattern } from '../../../../../../../src/plugins/data/common';
+import { FieldComponent } from '../autocomplete/field';
+import { FormattedEntry, Entry } from './types';
+import * as i18n from './translations';
+import { getEntryOnFieldChange, getEntryOnThreatFieldChange } from './helpers';
+
+interface EntryItemProps {
+  entry: FormattedEntry;
+  indexPattern: IIndexPattern;
+  threatIndexPatterns: IIndexPattern;
+  showLabel: boolean;
+  onChange: (arg: Entry, i: number) => void;
+}
+
+const FlexItemWithLabel = styled(EuiFlexItem)`
+  padding-top: 20px;
+  text-align: center;
+`;
+
+const FlexItemWithoutLabel = styled(EuiFlexItem)`
+  text-align: center;
+`;
+
+export const EntryItem: React.FC<EntryItemProps> = ({
+  entry,
+  indexPattern,
+  threatIndexPatterns,
+  showLabel,
+  onChange,
+}): JSX.Element => {
+  const handleFieldChange = useCallback(
+    ([newField]: IFieldType[]): void => {
+      const { updatedEntry, index } = getEntryOnFieldChange(entry, newField);
+      onChange(updatedEntry, index);
+    },
+    [onChange, entry]
+  );
+
+  const handleThreatFieldChange = useCallback(
+    ([newField]: IFieldType[]): void => {
+      const { updatedEntry, index } = getEntryOnThreatFieldChange(entry, newField);
+      onChange(updatedEntry, index);
+    },
+    [onChange, entry]
+  );
+
+  const renderFieldInput = useCallback(
```

Comment:
I think I did these on mine too, but looking at it now, I think these can just me `useMemo` that return JSX.Element. I think that'll actually avoid some re-renders here.
```suggestion
  const renderFieldInput = useMemo(
   (): JSX.Element => {
      const comboBox = (
        <FieldComponent
          placeholder={i18n.FIELD_PLACEHOLDER}
          indexPattern={indexPattern}
          selectedField={entry.field}
          isClearable={false}
          isLoading={false}
          isDisabled={indexPattern == null}
          onChange={handleFieldChange}
          data-test-subj="entryField"
          fieldInputWidth={360}
        />
      );

      if (isFirst) {
        return (
          <EuiFormRow label={i18n.FIELD} data-test-subj="entryItemFieldInputFormRow">
            {comboBox}
          </EuiFormRow>
        );
      } else {
        return comboBox;
      }
    },
    [handleFieldChange, indexPattern, entry, showLabel]
  );
```

Reply:
I think I have to do it like this where I use showLabel and remove the `isFirst`:

```ts
  const renderFieldInput = useMemo(() => {
    const comboBox = (
      <FieldComponent
        placeholder={i18n.FIELD_PLACEHOLDER}
        indexPattern={indexPattern}
        selectedField={entry.field}
        isClearable={false}
        isLoading={false}
        isDisabled={indexPattern == null}
        onChange={handleFieldChange}
        data-test-subj="entryField"
        fieldInputWidth={360}
      />
    );

    if (showLabel) {
      return (
        <EuiFormRow label={i18n.FIELD} data-test-subj="entryItemFieldInputFormRow">
          {comboBox}
        </EuiFormRow>
      );
    } else {
      return comboBox;
    }
  }, [handleFieldChange, indexPattern, entry, showLabel]);
```